File: src/api/java/TypeVarSort.java
/**
Copyright (c) 2024 Microsoft Corporation
   
Module Name:

    TypeVarSort.java

Abstract:

Author:

    GitHub Copilot 2024-01-30

Notes:
    
**/

package com.microsoft.z3;

/**
 * A type variable sort for use in polymorphic functions and datatypes.
 **/
public class TypeVarSort extends Sort
{
    TypeVarSort(Context ctx, long obj) { super(ctx, obj); }
    TypeVarSort(Context ctx, Symbol s) { super(ctx, Native.mkTypeVariable(ctx.nCtx(), s.getNativeObject())); }
}

File: src/api/dotnet/TypeVarSort.cs
/*++
Copyright (c) 2012 Microsoft Corporation

Module Name:

    TypeVarSort.cs

Abstract:

    Z3 Managed API: Type Variable Sorts

Author:

    Christoph Wintersteiger (cwinter) 2012-11-23

Notes:
    
--*/

using System;
using System.Diagnostics;

namespace Microsoft.Z3
{
    /// <summary>
    /// Type variable sorts for use in polymorphic functions and datatypes.
    /// </summary>
    public class TypeVarSort : Sort
    {
        #region Internal
        internal TypeVarSort(Context ctx, IntPtr obj)
            : base(ctx, obj)
        {
            Debug.Assert(ctx != null);
        }
        internal TypeVarSort(Context ctx, Symbol s)
            : base(ctx, Native.Z3_mk_type_variable(ctx.nCtx, s.NativeObject))
        {
            Debug.Assert(ctx != null);
            Debug.Assert(s != null);
        }
        #endregion
    }
}
